### ISSUE TITLE
Incompatibility with reticulate and rpy2

### ISSUE THREAD

**Issue by jeffwong-nflx** [rstudio/reticulate#321]
I have a code base that is in python and I am trying to integrate an R library into it. I am using `rpy2` to call an R function. 

I've found that if the R function calls reticulate, it will cause a segfault (core dumped). I think this is because of the circular nature of using Python to call R to call Python. However, I felt this use case might be common: for example an R developer might write a function `foo` that uses `sparse.model.matrix`, and passes that data to scipy through reticulate. A python developer that is trying to integrate the library into the application `bar` wants to call `foo` directly - this will cause the segfault. The developers could coordinate so that `foo` only returns the sparse matrix without calling scipy, and the application `bar` receives the sparse matrix and calls scipy itself. This is pretty challenging, so I'm wondering if there is something simpler I am missing to handle this edge case.

Here is a reproducible example of python calling R calling python.

```
from rpy2.robjects.packages import importr
import rpy2.robjects as robjects

r_func = robjects.r('''
  myfunc = function() {
    require(reticulate)
    sys = import("sys")
  }
  ''')
  
response = r_func()
```

```
*** Error in `python': double free or corruption (out): 0x00007fad7ccbf180 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7fad814317e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x8037a)[0x7fad8143a37a]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x4c)[0x7fad8143e53c]
```

**Comment by jjallaire** [comment]
No there isn't a way to create a secondary Python process.

We have spent some time trying to get this to work (https://github.com/rstudio/reticulate/issues/98) but so far to no avail.

**Comment by gokceneraslan** [comment]
There is a long discussion already in #208, so better to close this as duplicate I think.

**Comment by t-kalinowski** [comment]
Fixed via #1188
